source/common/http/conn_manager_impl.cc

```diff
@@ -285,6 +285,11 @@ void ConnectionManagerImpl::handleCodecException(const char* error) {
   read_callbacks_->connection().close(Network::ConnectionCloseType::FlushWriteAndDelay);
 }
 
+void ConnectionManagerImpl::forceCodecCreation() {
```

Comment:
Maybe just call this `createCodec`, have it take data as an argument, and use it internally as well (e.g. in `onData` below).

Reply:
Given right now there is only one I was going to leave this to a subsequent PR that introduced the ability to have multiple API listeners. Otherwise, I couldn't think of what other stats to expose at the moment.